msal_extensions/cache_lock.py

```diff
@@ -25,7 +32,29 @@ def __init__(self, lockfile_path):
             flags=portalocker.LOCK_EX | portalocker.LOCK_NB,
             **open_kwargs)
 
+    def try_to_create_lock_file(self):
+        timeout = 5
+        check_interval = 0.25
+        timeout_end = current_time() + timeout
+        while timeout_end > current_time():
+            try:
+                with open(self._lockpath, 'x'):
+                    return True
+            except OSError as err:
+                if err.errno == errno.EEXIST:
+                    logger.warning("Lock file exists, trying again after some time")
+                    time.sleep(check_interval)
+                else:
+                    raise
+            except ValueError:
+                logger.warning("Python 2 does not support atomic creation of file")
+                return False
```

Comment:
On second thought, since we are already excluding Python 2 anyway, we do not have to use all those `OSError`, `error.EEXIST`, and the tricky `else: raise`. The following might just work. Mind give it a shot?

```suggestion
            except ValueError:  # This needs to be the first clause, for Python 2 to hit it
                logger.warning("Python 2 does not support atomic creation of file")
                return False
            except FileExistsError:  # Only Python 3 will reach this clause
                logger.warning("Lock file exists, trying again after some time")
                time.sleep(check_interval)
```